idp-binding-service/src/main/java/io/mosip/idp/binding/services/WalletBindingServiceImpl.java

```diff
@@ -0,0 +1,241 @@
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import io.mosip.idp.binding.dto.BindingTransaction;
+import io.mosip.idp.binding.entity.IdTokenMapping;
+import io.mosip.idp.binding.entity.PublicKeyRegistry;
+import io.mosip.idp.binding.repository.IdTokenMappingRepository;
+import io.mosip.idp.binding.repository.PublicKeyRegistryRepository;
+import io.mosip.idp.core.dto.AuthChallenge;
+import io.mosip.idp.core.dto.KycAuthRequest;
+import io.mosip.idp.core.dto.KycAuthResult;
+import io.mosip.idp.core.dto.WalletBindingRequest;
+import io.mosip.idp.core.dto.WalletBindingResponse;
+import io.mosip.idp.core.exception.IdPException;
+import io.mosip.idp.core.exception.InvalidAuthFactorTypeException;
+import io.mosip.idp.core.exception.InvalidIndividualIdException;
+import io.mosip.idp.core.exception.InvalidTransactionException;
+import io.mosip.idp.core.exception.KycAuthException;
+import io.mosip.idp.core.spi.AuthenticationWrapper;
+import io.mosip.idp.core.spi.WalletBindingService;
+import io.mosip.idp.core.util.Constants;
+import io.mosip.idp.core.util.IdentityProviderUtil;
+import io.mosip.kernel.signature.dto.JWTSignatureRequestDto;
+import io.mosip.kernel.signature.dto.JWTSignatureResponseDto;
+import io.mosip.kernel.signature.service.SignatureService;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@Service
+public class WalletBindingServiceImpl implements WalletBindingService {
+
+	@Autowired
+	private CacheUtilService cacheUtilService;
+
+	@Autowired
+	private AuthenticationWrapper authenticationWrapper;
+
+	@Autowired
+	private PublicKeyRegistryRepository publicKeyRegistryRepository;
+
+	@Autowired
+	private IdTokenMappingRepository idTokenMappingRepository;
+
+	private SignatureService signatureService;
+
+	private ObjectMapper objectMapper;
+
+	@Value("${mosip.idp.binding.auth-partner-id}")
+	private String authPartnerId;
+
+	@Value("${mosip.idp.binding.auth-license-key}")
+	private String apiKey;
+
+	@Value("${mosip.idp.binding.public-key-expire-days}")
+	private long expireDays;
+
+	@Value("${mosip.idp.binding.issuer-id}")
+	private String issuerId;
+
+	@Override
+	public void sendBindingOtp() throws IdPException {
+		// TODO Auto-generated method stub
+
+	}
+
+	@Override
+	public WalletBindingResponse bindWallet(WalletBindingRequest walletBindingRequest) throws IdPException {
+		BindingTransaction bindingTransaction = cacheUtilService
+				.getTransaction(walletBindingRequest.getTransactionId());
+		if (bindingTransaction == null)
+			throw new InvalidTransactionException();
+
+		if (!bindingTransaction.getIndividualId().equals(walletBindingRequest.getIndividualId()))
+			throw new InvalidIndividualIdException();
+
+		if (!walletBindingRequest.getAuthChallenge().equals(walletBindingRequest.getAuthChallenge()))
+			throw new InvalidAuthFactorTypeException();
+
+		KycAuthResult kycAuthResult = authenticateIndividual(bindingTransaction, walletBindingRequest);
+
+		byte[] salt = IdentityProviderUtil.generateSalt(16);
```

Comment:
let's take the salt length from the properties file.

Reply:
fixed